### PR TITLE
Add a SpawnerPage error alert when the server fails

### LINKED ISSUE #368: [Bug]: Make sure the server sends back errors when starting NB
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

#281 is a catch-all for the backend swallowing the errors and not returning them to the frontend. This issue is specifically for the Spawner page. We need to send back and error if we have one when we go to start up a NB... no feedback to the user when it fails is not good.

### Expected Behavior

When there is a resource conflict or some other issue on the server, we should present that error back to the client and show it next to the submit button on the Spawner page.

That way the user knows something is wrong, and at worst case can report the error effectively.

### Steps To Reproduce

The issue that caused this was actually an invalid username -- we didn't translate it for safe usage in labels or kube resource names. Since that has been fixed on the frontend, it may be hard to reproduce this.

The backend creates a notebook, and reads the corresponding route that is created. I'm not sure how we can mimic a new error state from live code.

There is a [code issue](https://github.com/opendatahub-io/odh-dashboard/blob/main/backend/src/routes/api/notebooks/notebookUtils.ts#L90) -- if we fail to create the notebook, there is an assumption the route will still be created but that is not true. What we need to do is prevent this code issue on the backend and render the corresponding error on the frontend for the user to consume.

### Workaround (if any)

_No response_

### OpenShift Version

_No response_

### Openshift Version

_No response_

### What browsers are you seeing the problem on?

_No response_

### Open Data Hub Version

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by andrewballantyne**
Resolves #368

## Description
We don't show errors to the user when something goes wrong on the server. This will add an alert above the submit button.

When an error happens, it will look something like this (the body of the alert is the kube generated error): 
image

## How Has This Been Tested?
No easy way to test it, I forced an error on the backend by changing code. 

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: frontend/src/pages/notebookController/screens/server/SpawnerPage.tsx [hunk 1]
```diff
@@ -445,24 +455,35 @@ const SpawnerPage: React.FC = React.memo(() => {
...
-              }}
-              isDisabled={createInProgress}
-            >
-              Start server
-            </Button>
-            <Button variant="secondary" onClick={() => history.push('/')}>
-              Cancel
-            </Button>
-          </ActionGroup>
+          <div>
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/notebookController/screens/server/SpawnerPage.tsx:458]: May I ask why we need this `div` here?

Comment (andrewballantyne) [reply frontend/src/pages/notebookController/screens/server/SpawnerPage.tsx:458]: Funky way the forms are built with spacing between each item, makes it have ungodly white spaces around everything if they don't share the same singular DOM node in the form.

I logged https://github.com/opendatahub-io/odh-dashboard/issues/396 to take this further, hopefully cleaning up this random `div` and make this an official form footer.

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

**Review by cfchase — APPROVED** [review]
lgtm

**Pull request merged**